src/connection/manager.js

```diff
@@ -99,6 +100,9 @@ class ConnectionManager {
       connection.theirPeerInfo.disconnect()
       this.switch.emit('peer-mux-closed', connection.theirPeerInfo)
```

Comment:
Should we remove the listener added to the connection above with `connection.once('muxed', ...)`?

Reply:
The connections will remove all listeners attached to them once they hit the 'disconnect' state. This is to try to and keep listener management as simple as possible.

Reply:
Cool that makes sense